src/inmanta/data/model.py

```diff
@@ -982,6 +983,52 @@ def has_source(self) -> bool:
         """Can this config get packages from anywhere?"""
         return bool(self.index_url) or self.use_system_config
 
+    def get_index_args(self) -> list[str]:
+        """
+        Returns the index-related arguments that should be used to run a pip command
+        with this pip config.
+        """
+        index_args: list[str] = []
+        if self.index_url:
+            index_args.append("--index-url")
+            index_args.append(self.index_url)
+        elif not self.use_system_config:
+            # If the config doesn't set index url
+            # and we are not using system config,
+            # then we need to disable the index.
+            # This can only happen if paths is also set.
+            index_args.append("--no-index")
+        for extra_index_url in self.extra_index_url:
+            index_args.append("--extra-index-url")
+            index_args.append(extra_index_url)
+        return index_args
+
+    def get_environment_variables(self) -> dict[str, str]:
+        """
+        Returns the environment variables that should be used to run a pip command
+        with this pip config.
+        """
+        sub_env = os.environ.copy()
+        if not self.use_system_config:
+            # If we don't use system config, unset env vars
+            if "PIP_EXTRA_INDEX_URL" in sub_env:
+                del sub_env["PIP_EXTRA_INDEX_URL"]
+            if "PIP_INDEX_URL" in sub_env:
+                del sub_env["PIP_INDEX_URL"]
+            if "PIP_PRE" in sub_env:
+                del sub_env["PIP_PRE"]
+            if "PIP_NO_INDEX" in sub_env:
+                del sub_env["PIP_NO_INDEX"]
```

Comment:
```suggestion
for key in ("PIP_EXTRA_INDEX_URL", "PIP_INDEX_URL", "PIP_PRE", "PIP_NO_INDEX"):
    sub_env.pop(key, None)
```
We can use pop instead of del to avoid the if check, either way, I think iterating over the keys is cleaner

Reply:
I just moved this method to a different class, so I didn't really evaluate the implementation. It's an improvement so I'll apply it anyway.